src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1226,4 +1228,33 @@ public boolean checkIfRequestForItemExistOrCreateAsynch(@NotNull final ItemStack
 
         return false;
     }
+
+    /**
+     * Try to transfer a item matching a predicate from a position to the cook.
+     *
+     * @param pos       the position to transfer it from.
+     * @param predicate the predicate to evaluate.
+     * @return true if succesful.
+     */
+    public boolean tryTransferFromPosToWorker(final BlockPos pos, @NotNull final Predicate<ItemStack> predicate)
+    {
+        final TileEntity entity = world.getTileEntity(pos);
+        if (entity instanceof TileEntityChest)
```

Comment:
Their is no reason to do this. Both entities are ICAPProviders with a IITemHandler. Please make this one method call.

Reply:
kk